src/core/core.datasetController.js

```diff
@@ -65,6 +135,15 @@ module.exports = function(Chart) {
 			this.update(true);
 		},
 
+		/**
+		 * @private
+		 */
+		destroy: function() {
+			if (this._data) {
+				unlistenArrayEvents(this._data, this);
+			}
+		},
+
 		createMetaDataset: function() {
```

Comment:
You right, we should document `datasetElementType` and `dataElementType`, not sure about `createMetaDataset` and `createMetaData`. That can be done in a future PR though :)